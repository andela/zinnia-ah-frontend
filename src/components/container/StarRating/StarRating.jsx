import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { Rating } from 'semantic-ui-react';

import { starRating } from '../../../store/modules/article';

class StarRating extends Component {
  handleRate = (e, { rating }) => {
    return this.props.starRating(rating);
  };

  componentDidMount() {
    return this.props.starRating(this.props.rating);
  }

  render() {
    return (
      <div>
        <Rating maxRating={5} onRate={this.handleRate} size="massive" />
      </div>
    );
  }
}

StarRating.propTypes = {
  starRating: PropTypes.func,
  rating: PropTypes.number,
};

const mapStateToProps = state => ({
  rating: state.article.rating,
  averageRating: state.article.averageRating,
});

export default connect(
  mapStateToProps,
  { starRating },
)(StarRating);
