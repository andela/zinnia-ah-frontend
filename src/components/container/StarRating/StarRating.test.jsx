import React from 'react';
import { mount } from 'enzyme';

import StarRating from './StarRating';

describe('<StarRating />', () => {
  it('should render correctly', () => {
    const component = mount(<StarRating />);
    expect(component).toMatchSnapshot();
  });
});
