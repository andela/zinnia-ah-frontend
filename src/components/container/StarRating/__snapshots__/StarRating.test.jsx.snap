// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StarRating /> should render correctly 1`] = `
<StarRating>
  <Rating
    clearable="auto"
    defaultRating="3"
    icon="star"
    maxRating="5"
  >
    <div
      className="ui star rating"
      onMouseLeave={[Function]}
      role="radiogroup"
      tabIndex={-1}
    >
      <RatingIcon
        active={true}
        aria-checked={false}
        aria-posinset={1}
        aria-setsize="5"
        as="i"
        index={0}
        key="0"
        onClick={[Function]}
        onMouseEnter={[Function]}
        selected={false}
        tabIndex={0}
      >
        <i
          aria-checked={false}
          aria-posinset={1}
          aria-setsize="5"
          className="active icon"
          onClick={[Function]}
          onKeyUp={[Function]}
          onMouseEnter={[Function]}
          role="radio"
          tabIndex={0}
        />
      </RatingIcon>
      <RatingIcon
        active={true}
        aria-checked={false}
        aria-posinset={2}
        aria-setsize="5"
        as="i"
        index={1}
        key="1"
        onClick={[Function]}
        onMouseEnter={[Function]}
        selected={false}
        tabIndex={0}
      >
        <i
          aria-checked={false}
          aria-posinset={2}
          aria-setsize="5"
          className="active icon"
          onClick={[Function]}
          onKeyUp={[Function]}
          onMouseEnter={[Function]}
          role="radio"
          tabIndex={0}
        />
      </RatingIcon>
      <RatingIcon
        active={true}
        aria-checked={false}
        aria-posinset={3}
        aria-setsize="5"
        as="i"
        index={2}
        key="2"
        onClick={[Function]}
        onMouseEnter={[Function]}
        selected={false}
        tabIndex={0}
      >
        <i
          aria-checked={false}
          aria-posinset={3}
          aria-setsize="5"
          className="active icon"
          onClick={[Function]}
          onKeyUp={[Function]}
          onMouseEnter={[Function]}
          role="radio"
          tabIndex={0}
        />
      </RatingIcon>
      <RatingIcon
        active={false}
        aria-checked={false}
        aria-posinset={4}
        aria-setsize="5"
        as="i"
        index={3}
        key="3"
        onClick={[Function]}
        onMouseEnter={[Function]}
        selected={false}
        tabIndex={0}
      >
        <i
          aria-checked={false}
          aria-posinset={4}
          aria-setsize="5"
          className="icon"
          onClick={[Function]}
          onKeyUp={[Function]}
          onMouseEnter={[Function]}
          role="radio"
          tabIndex={0}
        />
      </RatingIcon>
      <RatingIcon
        active={false}
        aria-checked={false}
        aria-posinset={5}
        aria-setsize="5"
        as="i"
        index={4}
        key="4"
        onClick={[Function]}
        onMouseEnter={[Function]}
        selected={false}
        tabIndex={0}
      >
        <i
          aria-checked={false}
          aria-posinset={5}
          aria-setsize="5"
          className="icon"
          onClick={[Function]}
          onKeyUp={[Function]}
          onMouseEnter={[Function]}
          role="radio"
          tabIndex={0}
        />
      </RatingIcon>
    </div>
  </Rating>
</StarRating>
`;
